### PR TITLE
Record api hit on dataset query

### PR DESCRIPTION
**Pull request by cameronmarsh**

### CODE REVIEW

File: apps/discovery_api/lib/discovery_api_web/utilities/query_access_utils.ex [hunk 1]
```diff
@@ -6,34 +6,38 @@ defmodule DiscoveryApiWeb.Utilities.QueryAccessUtils do
...
-        {:ok, session}
-
-      false ->
-        {:error, "Session not authorized"}
+    if user_can_access_models?(authorized_models, current_user) do
+      session_opts = DiscoveryApi.prestige_opts()
+      session = Prestige.new_session(session_opts)
+      {:ok, session}
+    else
+      # IO.inspect(authorized_models, label: "Unauthorized models")
```
Comment (LtChae) [inline apps/discovery_api/lib/discovery_api_web/utilities/query_access_utils.ex:17]: Commented code, and an inspect no less!

**Review by LtChae — APPROVED** [review]

File: apps/discovery_api/lib/discovery_api_web/utilities/query_access_utils.ex (hunk 1 above)
Comment (cameronmarsh) [reply apps/discovery_api/lib/discovery_api_web/utilities/query_access_utils.ex:17]: :-|

File: apps/discovery_api/lib/discovery_api_web/controllers/multiple_data_controller.ex [hunk 2]
```diff
@@ -5,11 +5,17 @@ defmodule DiscoveryApiWeb.MultipleDataController do
...
+    only: [dataset_query: 0]
+
   plug(:accepts, MultipleDataView.accepted_formats())
 
   def query(conn, _params) do
     with {:ok, statement, conn} <- read_body(conn),
-         {:ok, session} <- QueryAccessUtils.authorized_session(conn, statement) do
+         {:ok, affected_models} <- QueryAccessUtils.get_affected_models(statement),
+         {:ok, session} <- QueryAccessUtils.authorized_session(conn, affected_models) do
+      Brook.Event.send(DiscoveryApi.instance_name(), dataset_query(), __MODULE__, Enum.map(affected_models, &Map.get(&1, :id)))
```
Comment (JarredOlson) [inline apps/discovery_api/lib/discovery_api_web/controllers/multiple_data_controller.ex:17]: This needs to send multiple events about 1 dataset

File: apps/discovery_api/lib/discovery_api/event/event_handler.ex [hunk 3]
```diff
@@ -97,6 +105,19 @@ defmodule DiscoveryApi.Event.EventHandler do
     end
   end
 
+  def handle_event(%Brook.Event{type: dataset_query(), data: affected_model_ids, author: author, create_ts: timestamp}) do
+    Enum.each(affected_model_ids, fn model_id ->
```
Comment (JarredOlson) [inline apps/discovery_api/lib/discovery_api/event/event_handler.ex:109]: This needs to be 1 dataset per event not multiple

**Review by JarredOlson — CHANGES_REQUESTED** [review]

**Review by JarredOlson — APPROVED** [review]

**Pull request merged**